backend/dataall/modules/shares_base/services/sharing_service.py

```diff
@@ -254,27 +294,50 @@ def verify_share(
         Returns True when completed
         -------
         """
-        with engine.scoped_session() as session:
-            share_data, share_items = cls._get_share_data_and_items(session, share_uri, status, healthStatus)
-            for type, processor in ShareProcessorManager.SHARING_PROCESSORS.items():
-                try:
-                    log.info(f'Verifying permissions with {type.value}')
-                    shareable_items = ShareObjectRepository.get_share_data_items_by_type(
-                        session,
-                        share_data.share,
-                        processor.shareable_type,
-                        processor.shareable_uri,
-                        status=status,
-                        healthStatus=healthStatus,
+        task_exceptions = []
+        try:
+            with engine.scoped_session() as session:
+                health_status_list: List[bool] = []
+                share_data, share_items = cls._get_share_data_and_items(session, share_uri, status, healthStatus)
+                for type, processor in ShareProcessorManager.SHARING_PROCESSORS.items():
+                    try:
+                        log.info(f'Verifying permissions with {type.value}')
+                        shareable_items = ShareObjectRepository.get_share_data_items_by_type(
+                            session,
+                            share_data.share,
+                            processor.shareable_type,
+                            processor.shareable_uri,
+                            status=status,
+                            healthStatus=healthStatus,
+                        )
+                        if shareable_items:
+                            health_status = processor.Processor(
```

Comment:
you don't use the list so it might as well be a boolean `shares_health_status &= processor....`